pkg/reconciler/knativeserving/common/images.go

```diff
@@ -144,3 +145,11 @@ func updateContainer(container *corev1.Container, newImage string, log *zap.Suga
 func replaceName(imageTemplate string, name string) string {
 	return strings.ReplaceAll(imageTemplate, containerNameVariable, name)
 }
+
+func updatePodImagePullSecrets(deployment *appsv1.Deployment, registry *servingv1alpha1.Registry, log *zap.SugaredLogger) {
+	if len(registry.ImagePullSecrets) > 0 {
+		log.Debugf("Adding ImagePullSecrets: %v", registry.ImagePullSecrets)
+		existingSecrets := deployment.Spec.Template.Spec.ImagePullSecrets
+		deployment.Spec.Template.Spec.ImagePullSecrets = append(registry.ImagePullSecrets, existingSecrets...)
```

Comment:
Is there any particular reason that you put registry.ImagePullSecrets in front of the existingSecrets, not behind??

Reply:
I don't think it matters functionally because according to the docs all of the credentials from various sources (service account, pod spec) get merged.

Because I don't think it matters functionally, I primarily opted for this because the code felt clean to me. If I were to swap the order of arguments to `append()`, I would need an extra conditional because the first argument to `append()` can't be nil.

Reply:
I ended up changing this code after realizing that I was potentially mutating the passed in registry through my append call.